core/metrics-core-api/src/main/java/org/hawkular/metrics/core/api/MetricsService.java

```diff
@@ -144,6 +144,7 @@
      * @return An {@link Observable} that emits the results with the same ordering as funcs
      * @see Aggregate
      */
+    @SuppressWarnings("unchecked")
```

Comment:
I would like to avoid `@Safevargs` with final especially since this is an interface and implementing classes.